crates/storage/src/lazy/lazy_array.rs

```diff
@@ -289,7 +245,7 @@ where
     /// returns the old value if any.
     fn put(&self, at: Index, new_value: Option<T>) -> Option<T> {
         mem::replace(
-            unsafe { self.entries.as_slice()[at as usize].get_ptr().as_mut() },
+            unsafe { self.entries[at as usize].get_ptr().as_mut() },
```

Comment:
Not sure about if I can make this change. Hmm, at least everything builds and tests run.

If I include the `.as_slice()` I need `#![feature(array_methods)]`, otherwise I get build errors.